scripts/gen-i18n.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Regenerates the translations base file by walking the source tree and
+ * parsing each file with flow-parser. Emits a JSON file with the
+ * translatable strings mapped to themselves in the order they appeared
+ * in the files and grouped by the file they appeared in.
+ *
+ * Usage: node scripts/gen-i18n.js
+ */
+const fs = require('fs');
+const path = require('path');
+
+const walk = require('walk');
+
+const flowParser = require('flow-parser');
+const estreeWalker = require('estree-walker');
+
+const TRANSLATIONS_FUNCS = ['_t', '_td', '_tJsx'];
+
+// A selection of plural variants to put in the base file: other langauges
+// with more types of plural will have more, but they will just be in the file
```

Comment:
Maybe clarify that we only need to worry about English plurals here because the base file is English